### PR TITLE
Replace all Auto-discovery instances to Autodiscovery 

### PR DESCRIPTION
**Pull request by fermujica**
This gives consistency with the product's UI. 
See [DOCS-2565](https://www.mulesoft.org/jira/browse/DOCS-2565).

### CODE REVIEW

File: anypoint-studio/v/6.5/setting-up-your-development-environment.adoc [hunk 1]
```diff
@@ -114,7 +114,7 @@ Note that you can either import the certificate to your existing keytool or crea
 After configuring and enabling Private Cloud support you can:
 
 . Log in to your Anypoint Platform Private Cloud Edition from Studio configuring your platform's credentials following the steps link:/anypoint-studio/v/6.5/setting-up-your-development-environment#configure-anypoint-platform-credentials[described above].
-. Configure link:/anypoint-studio/v/6.5/api-sync-reference[API Sync] and link:/api-manager/api-auto-discovery[API Autodiscovery] for APIs in your Private Cloud Edition platform.
```
Comment (sleslie23) [inline anypoint-studio/v/6.5/setting-up-your-development-environment.adoc]: I see multiple link changes in this PR, but I don't see any filename changes. Do both files already exist?

Comment (fermujica) [reply anypoint-studio/v/6.5/setting-up-your-development-environment.adoc]: No, I saw a couple of false positives for the regex I used in the find and replace. I'm fixing those.

File: anypoint-studio/v/6.5/setting-up-your-development-environment.adoc [hunk 2]
```diff
@@ -114,7 +114,7 @@ Note that you can either import the certificate to your existing keytool or crea
...
+. Configure link:/anypoint-studio/v/6.5/api-sync-reference[API Sync] and link:/api-manager/api-Autodiscovery[API Autodiscovery] for APIs in your Private Cloud Edition platform.
```
Comment (mtqp) [inline anypoint-studio/v/6.5/setting-up-your-development-environment.adoc]: File change

File: api-manager/v/1.x/api-auto-discovery-reference.adoc [hunk 3]
```diff
@@ -13,11 +13,11 @@ The configuration.xml of a project that includes auto-discovery references the m
 
 The `api-platform-gw` global element contains the information needed to register an API in Anypoint Platform. The global element includes the description and tag attributes. The values you set for these attributes appear on the API version details page after registration:
 
-image::api-auto-discovery-reference-0dc02.png[api-auto-discovery-reference-0dc02]
+image::api-auto-discovery-reference-0dc02.png[api-Autodiscovery-reference-0dc02]
```
Comment (mtqp) [inline api-manager/v/1.x/api-auto-discovery-reference.adoc]: found another one

File: api-manager/v/1.x/api-auto-discovery.adoc [hunk 4]
```diff
@@ -30,19 +30,16 @@ Set system properties in Runtime Manager. In the property settings using the fol
...
 ** anypoint.platform.client_secret={your client secret}
 
-== Auto-Discovery Process
+== Autodiscovery Process
 
-The following diagram illustrates the auto-discovery process for a project containing an `api-platform-gw:api` auto-discovery global element in the configuration.xml:
+The following diagram illustrates the Autodiscovery process for a project containing an `api-platform-gw:api` Autodiscovery global element in the configuration.xml:
 
-image::api-auto-discovery-v2.png[api-auto-discovery-v2,height=452,width=600]
+image::api-auto-discovery-v2.png[api-Autodiscovery-v2,height=452,width=600]
```
Comment (mtqp) [inline api-manager/v/1.x/api-auto-discovery.adoc]: File name

File: api-manager/v/1.x/configure-auto-discovery-task.adoc [hunk 5]
```diff
@@ -57,7 +57,7 @@ If you are creating a new API, API Manager makes this name visible in Anypoint P
 +
 * In *API Version*, type an arbitrary version identifier, for example `1.0development`.
 +
-Do not change the version name later in API Portal, as explained in link:/api-manager/api-auto-discovery#best-practices["Best Practices"].
+Do not change the version name later in API Portal, as explained in link:/api-manager/api-Autodiscovery#best-practices["Best Practices"].
```
Comment (mtqp) [inline api-manager/v/1.x/configure-auto-discovery-task.adoc]: File

File: anypoint-studio/v/6.5/setting-up-your-development-environment.adoc (hunk 1 above)
Comment (fermujica) [reply anypoint-studio/v/6.5/setting-up-your-development-environment.adoc]: Ran an entire link checker against these sections. There are no other broken links around the auto-discovery link macros.

**Pull request merged**